src/System/specifics.rs

```diff
@@ -0,0 +1,390 @@
+                write!(f, "{}.{}", comp.name, location_id)
+            }
+            SpecificLocation::BranchLocation(left, right) => write!(f, "({}, {})", left, right),
+            SpecificLocation::SpecialLocation(spec) => write!(f, "{}", spec),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum SpecialLocation {
+    Universal,
+    Error,
+}
+
+impl fmt::Display for SpecialLocation {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SpecialLocation::Universal => write!(f, "[Universal]"),
+            SpecialLocation::Error => write!(f, "[Error]"),
+        }
+    }
+}
+
+impl SpecificState {
+    pub fn from_state(state: &State, sys: &dyn TransitionSystem) -> Self {
+        let locations = state_specific_location(state, sys);
+        let clock_map = specific_clock_comp_map(sys);
+
+        let constraints = state.zone_ref().minimal_constraints();
+        let constraints = SpecificDisjunction::from(constraints, &clock_map);
+        Self {
+            locations,
+            constraints,
+        }
+    }
+    pub fn from_state_pair(
+        state: &StatePair,
+        sys1: &dyn TransitionSystem,
+        sys2: &dyn TransitionSystem,
+    ) -> Self {
+        let locations = state_pair_specific_location(state, sys1, sys2);
+
+        let clock_map = specific_clock_comp_map2(sys1, sys2);
+
+        let constraints = state.ref_zone().minimal_constraints();
+        let constraints = SpecificDisjunction::from(constraints, &clock_map);
+        Self {
+            locations,
+            constraints,
+        }
+    }
+}
+
+impl fmt::Display for SpecificState {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let locs = &self.locations;
+
+        write!(f, "({})", locs)
+        // TODO: maybe show constraints
+        // write!(f, "({} | {})", locs, self.constraints)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct SpecificClock {
+    pub name: String,
+    pub id: ClockIndex,
+    pub comp: SpecificComp,
+}
+
+impl SpecificClock {
+    pub fn new(name: String, id: ClockIndex, comp: SpecificComp) -> Self {
+        Self { name, id, comp }
+    }
+}
+
+pub fn specific_clock_comp_map(sys: &dyn TransitionSystem) -> HashMap<ClockIndex, SpecificClock> {
+    sys.comp_infos()
+        .iter()
+        .flat_map(|comp| {
+            //let name = comp.name.clone();
+            //let id = comp.id;
+            comp.declarations
+                .clocks
+                .iter()
+                .map(move |(clock, &clock_id)| {
+                    (
+                        clock_id,
+                        SpecificClock::new(
+                            clock.clone(),
+                            clock_id,
+                            SpecificComp::new(comp.name.clone(), comp.id),
+                        ),
+                    )
+                })
+        })
+        .collect()
+}
+
+pub fn specific_clock_comp_map2(
```

Comment:
Maybe rename this to show that it is a map from multiple systems? `specific_clock_comp_map_composite`?